### PR TITLE
feat(rome_css_parser): CSS parser #4682

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

Initial implementation of `CssTokenSource` and `CssParser` for the CSS parser. 
Added `CssParserOptions` which has an `allow_single_line_comments` option. This allows the writing of single line comments in CSS syntax.

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/rome_css_parser/src/syntax.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use crate::parser::CssParser;
+use rome_css_syntax::CssSyntaxKind::*;
+use rome_parser::Parser;
+
+pub(crate) fn parse_root(p: &mut CssParser) {
```
Comment (denbezrukov) [inline crates/rome_css_parser/src/syntax.rs:5]: It's the entry point for syntax parsing.

File: crates/rome_css_parser/tests/spec_test.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+use std::path::Path;
+
+#[derive(Copy, Clone)]
+pub enum ExpectedOutcome {
+    Pass,
+    Fail,
+    Undefined,
+}
+
+pub fn run(test_case: &str, _snapshot_name: &str, test_directory: &str, outcome_str: &str) {
```
Comment (denbezrukov) [inline crates/rome_css_parser/tests/spec_test.rs:19]: It's a copy of the JSON test spec.

I'd like to write tests the same way we use in the JS parser. But I'm unsure what is the best way. I think that we can combine both approaches.

File: crates/rome_json_parser/src/parser.rs [hunk 3]
```diff
@@ -68,26 +66,4 @@ impl<'source> Parser for JsonParser<'source> {
     fn source_mut(&mut self) -> &mut Self::Source {
         &mut self.source
     }
-
-    /// Try to eat a specific token kind, if the kind is not there then adds an error to the events stack.
-    fn expect(&mut self, kind: Self::Kind) -> bool {
```
Comment (denbezrukov) [inline crates/rome_json_parser/src/parser.rs:73]: It seems that we don't need any special behavior for comments here. When `allow_comments` is true, it implies that `COMMENT` and `MULTILINE_COMMENT` kinds are considered as trivia.
https://github.com/rome/tools/blob/1dcd62b169e875b7048f63efb5f6636d2152334c/crates/rome_json_parser/src/token_source.rs#L62-L71

File: crates/rome_css_parser/tests/spec_test.rs (hunk 2 above)
Comment (ematipico) [reply crates/rome_css_parser/tests/spec_test.rs:19]: Personally, I don't like the test suite of the JS parser:
- we have to run two commands in order to run the tests
- we always need to tweak the test generator parser to accommodate new things - like parser options, which is becoming quite difficult

The best way, personally, would be just to write a file to test and then check the snapshot with `insta`; of course the testing suite will do much more, but that's the gist of it.

**Review by ematipico — APPROVED** [review]

**Pull request merged**